src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -772,13 +772,15 @@ func (c controller) waitForOLMOperators() bool {
 		}
 
 		operatorStatus := utils.CsvStatusToOperatorStatus(string(csv.Status.Phase))
-		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operator.Name, operatorStatus, csv.Status.Message)
-		if err != nil {
-			c.log.WithError(err).Warnf("Failed to update olm %s status", operator.Name)
-			continue
-		}
+		if operator.Status != operatorStatus || (operator.StatusInfo != csv.Status.Message && csv.Status.Message != "") {
```

Comment:
why we need the "&& csv.Status.Message != " in the condition? I think that we will skip updates - I guess that the default for csv.Status.Message is an empty string

Reply:
In case the message is empty there is nothing to update